From fdee25ac96d019b30012671da8a04e72b9861848 Mon Sep 17 00:00:00 2001
From: Alexander Filonenko <alexf@motorola.com>
Date: Wed, 12 Nov 2014 15:15:11 -0600
Subject: [PATCH 318/959] IKKRNBSP-3424 arm64: cpuinfo: show device and radio

Show Device and Radio in cpuinfo.
Port of IKXLUPGRD-5618

Change-Id: Ifbef2d7338d40d476ac46304fdcf19c5a085ea8b
Signed-off-by: Alexander Filonenko <alexf@motorola.com>
Reviewed-on: http://gerrit.mot.com/688653
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Jeffrey Carlyle <jeff.carlyle@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
Reviewed-on: http://gerrit.mot.com/710129
Reviewed-by: Jason Hrycay <jason.hrycay@motorola.com>
Submit-Approved: Jason Hrycay <jason.hrycay@motorola.com>
Tested-by: Jason Hrycay <jason.hrycay@motorola.com>
Signed-off-by: Ravi Vembu <raviv@motorola.com>
Reviewed-on: http://gerrit.mot.com/787266
SME-Granted: SME Approvals Granted
---
 arch/arm64/kernel/setup.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/kernel/setup.c b/arch/arm64/kernel/setup.c
index d8ed9db73ab..207e11ad5fe 100644
--- a/arch/arm64/kernel/setup.c
+++ b/arch/arm64/kernel/setup.c
@@ -61,6 +61,10 @@
 #include <asm/psci.h>
 #include <asm/efi.h>
 
+#include <soc/qcom/bootinfo.h>
+
+void __attribute__((weak)) mach_cpuinfo_show(struct seq_file *m, void *v);
+
 unsigned int processor_id;
 EXPORT_SYMBOL(processor_id);
 
@@ -507,6 +511,13 @@ static int c_show(struct seq_file *m, void *v)
 	else
 		seq_printf(m, "Hardware\t: %s\n", arch_read_hardware_id());
 
+	seq_printf(m, "Revision\t: %04x\n", system_rev);
+	seq_printf(m, "Serial\t\t: %08x%08x\n",
+		 system_serial_high, system_serial_low);
+
+	if (mach_cpuinfo_show)
+		mach_cpuinfo_show(m, v);
+
 	return 0;
 }
 
-- 
2.11.0

